src/proxy.rs

```diff
@@ -502,11 +527,129 @@ fn handle_server_event(
             );
 
             // TODO(#145): Fix unwrap
-            let handle = client_to_proxy.idle_reject(status).unwrap();
+            let handle = client_to_proxy.state.idle_reject(status).unwrap();
             trace!(role = "p2c", ?handle, "idle_reject");
         }
         client::Event::IdleDoneSent { handle } => {
             trace!(role = "p2s", ?handle, "--->");
         }
     }
 }
+
+struct ImapState<S> {
+    input_role: &'static str,
+    input_direction: &'static str,
+    input_color: Option<Color>,
+    input_fragmentizer: Fragmentizer,
+    output_role: &'static str,
+    output_direction: &'static str,
+    output_color: Option<Color>,
+    output_fragmentizer: Fragmentizer,
+    state: S,
+}
+
+impl ImapState<Client> {
+    pub fn client(state: Client) -> Self {
+        Self {
+            input_role: "s2p",
```

Comment:
Thought: An `enum Role` could determine `input_{role,direction,color}`? Something for a later cleanup.

Reply:
I... already did this. But it created a lot of noise because I changed all occurrences of `"s2p"`, `"p2s"`, `"c2p"` and `"p2c"`. Many single-line log statements became multi-line.

I would say let's do this later.